extensions/BugModal/web/bug_modal.js

```diff
@@ -1328,12 +1328,161 @@ $(function() {
             saveBugComment(event.target.value);
         });
 
+    function smartLinkPreviews() {
+        const getHrefIdParam = anchor => (new URL(anchor.href)).searchParams.get("id");
+        const filterUnique = (value, index, array) => value && array.indexOf(value) === index;
+
+        const findLinkElements = pathname => {
+            return (
+                Array
+                .from(document.querySelectorAll('.comment-text a'))
+                .filter(anchor => {
+                    return (
+                        anchor.hostname === window.location.hostname &&
```

Comment:
Is there a better way to get the current bmo instance hostname?

Reply:
Look at js_BUGZILLA in template/en/default/global/header.html.tmpl,
you can add Bugzilla->localconfig->{urlbase} to that structure.